lib/folio_client.rb

```diff
@@ -276,8 +286,9 @@ def with_token_refresh_when_unauthorized
     response = yield
 
     # if unauthorized, token has likely expired. try to get a new token and then retry the same request(s).
-    if response.status == 401
-      config.token = Authenticator.token(config.login_params, connection)
+    if response.status == 401 || response.status == 403
```

Comment:
FOLIO docs say that with Poppy release onward, if you send an expired token, it will return a 403 rather than a 401.